### PR TITLE
P4 2184 calculate standard moves

### PR DESCRIPTION
**Pull request by jcavell**
@matt-crowson-moj  apologies for another huge PR.

Still a bit of a WIP, but this now shows calculation of prices if you use the active profile of localstack and POST to the endpoint of http://localhost:8080/calculate-prices

Also did some refactoring around the locking / timing, which I moved to a single place in the ImportService (it was previously in multiple places).

I need to improve tests and clean up the code a bit.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/PriceCalculator.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import uk.gov.justice.digital.hmpps.pecs.jpc.reporting.MovePersonJourneysEvents
+
+@Component
+class PriceCalculatorFactory(
+        @Autowired val locationRepository: LocationRepository,
+        @Autowired val priceRepository: PriceRepository) {
+
+    fun calculator(moves: List<MovePersonJourneysEvents>): PriceCalculator {
+        val nomisAgencyId2Location = locationRepository.findAll().associateBy(Location::nomisAgencyId)
+        val (sercoPrices, geoPrices) = priceRepository.findAll().partition { it.supplier == Supplier.SERCO }
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/PriceCalculator.kt:21]: Not sure.  Perhaps load the prices in the `fun standardPrices(supplier: Supplier): List<MovePrice>` when needed. Dunno.

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/PriceCalculator.kt (hunk 1 above)
Comment (jcavell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/calculator/PriceCalculator.kt:21]: I did previously do that but I felt it was unclear when things were initialised / loaded in.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/StagingImporterTest.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.boot.test.context.SpringBootTest
+
+import org.springframework.test.context.ActiveProfiles
+import uk.gov.justice.digital.hmpps.pecs.jpc.pricing.Supplier
+
+import java.time.LocalDate
+
+@ActiveProfiles("dev")
+@SpringBootTest
+internal class StagingImporterTest {
```
[the remark this answers is not in the document]
Comment (jcavell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/reporting/StagingImporterTest.kt:17]: Interestingly, the dev profile uses the S3 creds set by environment variables. Yes it's a bit confusing.

**Pull request merged**